src/datasources/balances-api/balances-api.manager.ts

```diff
@@ -40,11 +43,31 @@ export class BalancesApiManager implements IBalancesApiManager {
     this.zerionBalancesApi = zerionBalancesApi;
   }
 
-  async getBalancesApi(chainId: string): Promise<IBalancesApi> {
+  async getBalancesApi(
+    chainId: string,
+    safeAddress: `0x${string}`,
+  ): Promise<IBalancesApi> {
     if (this.zerionChainIds.includes(chainId)) {
       return this.zerionBalancesApi;
     }
 
+    try {
+      const transactionApi =
+        await this.transactionApiManager.getTransactionApi(chainId);
+      await transactionApi.getSafe(safeAddress);
+      return this._getSafeBalancesApi(chainId);
+    } catch (err) {
+      return this.zerionBalancesApi;
+    }
+  }
+
+  async getFiatCodes(): Promise<string[]> {
+    const zerionFiatCodes = await this.zerionBalancesApi.getFiatCodes();
+    const safeFiatCodes = await this.coingeckoApi.getFiatCodes();
```

Comment:
We could make these concurrent. What do you think?
```suggestion
    const [zerionFiatCodes, safeFiatCodes] = await Promise.all([
      this.zerionBalancesApi.getFiatCodes(),
      this.coingeckoApi.getFiatCodes(),
    ]);
```

Reply:
Absolutely!